### DISCUSSION TITLE
Adding a setFilterSlidedownColumn settings...

### DISCUSSION THREAD

**Discussion by dgillier** [rappasoft/laravel-livewire-tables#1241] [Feature Requests]
Hello,

Will be great  to pick in which column a filter is displayed...
For exemple to stack up two selects in a column while a multiselect is using a lot of space in a previous column.

This cannot be done with the current setFilterSlidedownRow...(or maybe I'm missing something)

![Screenshot 2023-06-03 at 15 08 45](https://github.com/rappasoft/laravel-livewire-tables/assets/37960204/137ad114-d050-4a27-afa6-7bc022960543)

Thanks for considering :)
Denis

**Comment by lrljoe** [comment]
Did I add in a set filter slide down column function?

If not, I'll add it in.

Otherwise, it uses this order of the filters in your filters() to determine the order.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: dgillier ***@***.***>
Sent: Saturday, June 3, 2023 2:12:26 PM
To: rappasoft/laravel-livewire-tables ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [rappasoft/laravel-livewire-tables] Adding a setFilterSlidedownColumn settings... (Discussion #1241)


Hello,

Will be great to pick in which column a filter is displayed...
For exemple to stack up two selects in a column while a multiselect is using a lot of space in a previous column.

This cannot be done with the current setFilterSlidedownRow...(or maybe I'm missing something)

[Screenshot 2023-06-03 at 15 08 45]<https://user-images.githubusercontent.com/37960204/243094181-137ad114-d050-4a27-afa6-7bc022960543.png>

Thanks for considering :)
Denis

**Reply by lrljoe** [reply]
So I did add an option for colspan for a filter which you can use to expand the width of one or more filters across multiple columns.


```php
    setFilterSlidedownColspan(string $filterSlidedownColspan)
```

Otherwise it does just stick them in the order they're listed in the filters() with a default colspan of 1.

The problem I can foresee with specifying a column for the filter is I'd have to decide which wins when it's colspan and specified column, and the way it should behave, and code that in.

E g. Two filters both set to row 1.
One filter is set to a colspan of 3
The next filter  has a set column of 2
which one should win the space?

That and some columns don't exist at lower resolutions (e.g. 5 columns on desktop becomes 3 on tablet becomes 1 on mobile)

Happy to hear suggestions!

**Comment by dgillier** [comment]
hummm, yeah did not think about that...

One option is to do something similar as col-start-x...

or maybe to  use css 'columns 'instead of 'grid' which make it easier to do 'masonry' style.

What do you think ?

**Reply by lrljoe** [reply]
I'll take a look at it and see what I can come up with.

I had planned for the popover filter to have two columns too, so will probably do that at the same time.

**Reply by lrljoe** [reply]
I've slacked on this, apologies!  I'll review next week.

**Comment by lrljoe** [comment]
Still thinking this one through and how best to do it

If it was just tailwind it'd be easy enough, but having to try and think of an efficient way for both tailwind and bootstrap 4+5
